torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,1096 @@ namespace fuser {
 namespace cuda {
 namespace vectorize_helper {
 
-// Grab all values and expressions used to make the merged_domain and remove
-// them from the fusion
-void cleanUpInnermostMergedDomains(
-    const std::vector<IterDomain*>& root_domain,
-    IterDomain* merged_domain) {
-  TORCH_INTERNAL_ASSERT(merged_domain != nullptr);
-  TORCH_INTERNAL_ASSERT(!root_domain.empty());
+namespace factorization_helpers {
 
-  std::unordered_set<Val*> root_set({root_domain.begin(), root_domain.end()});
-
-  auto vals = DependencyCheck::getAllValsBetween(root_set, {merged_domain});
+std::multiset<int64_t> computeFactors(int64_t i) {
+  TORCH_INTERNAL_ASSERT(
+      i > 0, "Only support values for factorization >0 but received, ", i);
+  std::multiset<int64_t> factors;
+  // Take out multiples of two first as that's easy
+  while (i % 2 == 0) {
+    factors.emplace(2);
+    i /= 2;
+  }
 
-  for (auto it = vals.rbegin(); it != vals.rend(); ++it) {
-    TORCH_INTERNAL_ASSERT((*it)->isA<IterDomain>());
-    auto id = (*it)->as<IterDomain>();
-    if (root_set.find(id) != root_set.end()) {
-      continue;
+  // Brute force factorization:
+  bool updated = true;
+  while (updated && i > 1) {
+    updated = false;
+    for (auto factor = 3; factor * factor <= i; factor += 2) {
+      if (i % factor == 0) {
```

Comment:
should this `if` be a `while`?

Reply:
For example, if I have `i = 3*3*3*5*5*5`, this algorithm will find 9, 15 as a factor, but don't we want to have just 3 and 5?

Reply:
Thanks, missed a break 😳

Reply:
Yeah, that was not written well... thanks for flagging it.